src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -396,17 +349,7 @@ void IODataBatchOperationListener.BatchOperationContentStreamDisposed()
         /// This listener can choose to fail, if the currently written payload doesn't support in-stream error at this position.
         /// If the listener returns, the writer should not allow any more writing, since the in-stream error is the last thing in the payload.
         /// </remarks>
-        void IODataOutputInStreamErrorListener.OnInStreamError()
-        {
-            this.rawOutputContext.VerifyNotDisposed();
-            this.SetState(BatchWriterState.Error);
-            this.rawOutputContext.TextWriter.Flush();
-
-            // The OData protocol spec did not defined the behavior when an exception is encountered outside of a batch operation. The batch writer
-            // should not allow WriteError in this case. Note that WCF DS Server does serialize the error in XML format when it encounters one outside of a
-            // batch operation.
-            throw new ODataException(Strings.ODataBatchWriter_CannotWriteInStreamErrorForBatch);

```

Comment:
Common logic, especially state handling, should be done in base class implementation. Both ODataRawOutputContext and ODataJsonOutputContextBase have Flush methods that call flush on the underlying rawValueWriter/JsonWriter, so if you can add Flush to the ODataOutputContext base class you can just call outputContext.Flush() instead of this.rawOutputContext.TextWriter.Flush() in the MultipartMixed derived class, or this.JsonLightOutputContext.JsonWriter.Flush() in the JsonLight derived class.